### ISSUE TITLE
[FR] drilling down until no sub-categories should switch to part view.

### ISSUE THREAD

**Issue by method1020** [inventree/InvenTree#4328]
### Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find a similar feature request

### Problem statement

Drilling down into a sub-category until there are no more. Operators have to click on part view to see the parts in that sub-category, some users here find that confusing.

### Suggested solution

Once a user has drilled down through the category hierarchy the view should switch to part view instead of showing an empty category view.

### Describe alternatives you've considered

A notification that there are no more sub-categories. The 'no matching results' notification is confusing to some users.

### Do you want to develop this?

- [ ] I want to develop this.

**Comment by matmair** [comment]
@method1020 how do you imagine creating new categorie trees then? Always switching to the categories tag?

**Comment by MR-DOS** [comment]
What if the "Create category" button was also in the Part view?
For me - this is a rather unpleasant thing in Inventree that it does divide so strictly between categories and Parts. Having the category section on top (only the next lower level, and possibly collapsible) and Part beneath it would save so much clicking. There's probably a reason why most e-shops (I know, Inventree isn't an e-shop but operation-wise, these have some similarities) show the "category navigation bar" on top (as InvenTree also does), but also show the subcategories on the same page. 

To quote an unnamed e-shop:
![image](https://user-images.githubusercontent.com/29711626/218335079-8f0198f5-86c2-447b-9a65-9801957cb701.png)

**Comment by matmair** [comment]
InvenTree is so much more than an e-shop. Users in my social circle mostly use it as a parts library and for manufacturing coordination. It is basically a maintained version of PartKeepr for most users I know.
Mixing categories and parts will lead to unneeded confusion - I am against it without strong arguments for it and good help texts that help new users understand how parts, categories, categories parameters and part parameters work. There are 4 tabs in the categories view that would need to be integrated.

**Comment by miggland** [comment]
This is something which confuses me as well sometimes, it does lead to unnecessary clicks.

A relatively small change would be to change the text, maybe add a link?

Eg: "No matching _category_ results, [link]switch to part view[/link]"

![image](https://user-images.githubusercontent.com/1314853/218395312-67ff70da-23cb-4441-b9d2-7b2c4b5f62b5.png)

**Comment by method1020** [comment]
@miggland 

Rephrasing in my head would solve the issue here of confusion. A link to part view might also help:

![image](https://user-images.githubusercontent.com/3278824/218468202-2bdc9122-5b90-4552-8284-89c8d21146c4.png)

(I don't know how to add links!)

**Comment by SergeoLacruz** [comment]
Yes, sometimes users are confused. But it takes only a few days to get used to it. After I explained to five guys, the problem was gone :-)

**Comment by miggland** [comment]
Has @method1020's solution been implemented? That would solve the issue, IMO (even without links)

**Comment by MR-DOS** [comment]
No, it wasn't by 0.10.1. May test on 0.11.0 later.
![image](https://user-images.githubusercontent.com/29711626/232921760-62d86e47-df36-4bb3-b412-7230b7a47e63.png)

**Comment by matmair** [comment]
I am not aware that anyone ever worked on this and nothing is linked so it is reasonable to think that the suggested solution was not implemented.

**Comment by MR-DOS** [comment]
Confirmed, 0.11.0 still the same
![image](https://user-images.githubusercontent.com/29711626/233049083-da274523-9124-450d-bf1e-125b56b1b670.png)
